### ISSUE TITLE
Vectorization targets (pytorch)

### ISSUE THREAD

**Issue by CemTunaboylu** [CemTunaboylu/quantization#1]
Main Bottlenecks at the moment (in PyTorch):
- [ ] `reduce` creates a loop, thus not vectorized, use `torch.stack(...).sum(dim=0)` or  `torch.stack(...).bitwise_or(dim=0)` if possible instead
- [ ] Investigate elimination of if else branches to optimize, e.g. aget_max_bits_needed` in `weight_pack.py`
